### PR TITLE
[5710] Add `slugid` to DQT `trn_request` calls

### PR DESCRIPTION
**Pull request by stevehook**
### Context
We have a plan to use `slug` values to identify DQT records in order to help match trainee records more accurately between Register and DQT. This PR deals with sending the `Trainee#slug` to DQT service when we create a new trainee record in Register.

### Changes proposed in this pull request
We send the `slugid` attribute in requests to the `/v2/trn-requests/...` (request a TRN) DQT endpoint.

We also set the `Trainee#slug_sent_to_dqt_at` attribute so that we can track whether or not DQT has been sent the `slug` for a given trainee record.

This PR depends on https://github.com/DFE-Digital/register-trainee-teachers/pull/3447 and will remain in draft until that gets merged.

### Guidance to review
- Is there anything missing?
- Does this need to be behind a feature flag? (I've assumed not but can change that).

### Important business

- [x] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [x] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [x] Do we need to send any updates to DQT as part of the work in this PR?
- [x] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: app/services/trainees/set_slug_sent_at.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+module Trainees
+  class SetSlugSentAt
+    include ServicePattern
+
+    def initialize(trainee:)
+      @trainee = trainee
+    end
+
+    def call
+      trainee.slug_sent_to_dqt_at ||= Time.zone.now
```
Comment (defong) [inline app/services/trainees/set_slug_sent_at.rb:12]: Shouldn't it set it every time regardless? Feels odd that we only record once.

**Review by carlosmartinez — APPROVED** [review]

File: app/services/trainees/set_slug_sent_at.rb (hunk 1 above)
Comment (stevehook) [reply app/services/trainees/set_slug_sent_at.rb:12]: @defong originally I was going to just make this a boolean but then I thought that we would normally record the time that a thing happened. So this isn't a direct requirement and therefore it's totally open to debate! My thinking was that we are most likely to care about when we first synched slugs with DQT (e.g. if you ever needed to know whether DQT had the slug at a particular date in the past).

Comment (defong) [reply app/services/trainees/set_slug_sent_at.rb:12]: Oh right I was thinking we have like the timeline and the audit, hence the oddity.

I think it maybe better to have it set every time, so we can say we set it on the firstly and subsequently on theses dates based on audit and last recorded date.

WDYT?

**Review by defong — APPROVED** [review]
You the best :muscle:

**Pull request merged**